### ISSUE TITLE
GraphQL Code Generator fails on DateTime types

### ISSUE THREAD

**Issue by breyed** [EntityGraphQL/EntityGraphQL#119]
When [GraphQL Code Generator](https://www.graphql-code-generator.com/) reports the following error when the data context has a type with a property of type `DateTime`.

> Could not obtain introspection result, received: {"errors":[{"message":"Field '__schema' - No schema type found for dotnet type DateTimeOffset. Make sure you add it or add a type mapping"}]}

I'm using EntityGraphQL to expose an Entity Framework data context via ASP.NET Core like this:
``` c#
services.AddGraphQLSchema<GraphQLContext>(); // My GraphQLContext context has a type with a property of type DateTime.
...
app.UseEndpoints(endpoints => endpoints.MapGraphQL<GraphQLContext>());
```

This is the GraphQL Code Generator's `codegen.yml`:
``` yaml
overwrite: true
schema: "http://localhost:30397/graphql"
generates:
  src/generated/graphql.ts:
    plugins:
      - "typescript"
      - "typescript-operations"
      - "typescript-react-apollo"
```

There is no DateTimeOffset type in any of the data context types. My guess is that the [DateTimeOffset implicit conversion](https://docs.microsoft.com/en-us/dotnet/api/system.datetimeoffset.op_implicit?view=net-6.0#system-datetimeoffset-op-implicit(system-datetime)-system-datetimeoffset) from DateTime is causing the problem.

I tried creating a unit test for the problem. [This commit](https://github.com/breyed/EntityGraphQL/commit/0e6a8c0e44294d41dc87eb207d91b5e162dd94d4) has the test. It includes the query that GraphQL Code Generator uses. However, the result doesn't contain any information about the query beyond its name, so I don't think I'm setting up the test correctly.

**Comment by lukemurray** [comment]
Are you using `DateTimeOffset` in your `GraphQLContext`? If so you will need to tell the schema about it. You can either use it as a scalar

`baseSchema.AddScalarType<TimeSpan>("TimeSpan", "Represents a time interval or the time of day");`

And it will just treat it as a single value field. EntityGraphQL does this by default for `DateTime`

`schematTypes.Add("Date", new SchemaType<DateTime>(this, "Date", "Date with time scalar", null, false, false, true));`

**Comment by breyed** [comment]
I'm not using DateTimeOffset in the GraphQLContext or anywhere. Lots of DateTime types, but no DateTimeOffset, so the error was a surprise.

**Comment by lukemurray** [comment]
Ok, thanks.

Using Dotnet 6 & EF Core 6?

**Comment by breyed** [comment]
I'm using EF Core 6. I figured out the problem. It turns out that the data context did contain a `DateTimeOffset`, just not in my code. The implicit conversion was a red hearing. The real problem is that I'm using Microsoft.AspNetCore.Identity.EntityFrameworkCore, and one of my entities derives from `IdentityUser`, which has property `LockoutEnd` of type `DateTimeOffset?`. Since it's not my code, it didn't show up when I did a project-wide search for "DateTimeOffset". Unfortunately, GraphQL Code Generator doesn't show where the type was referenced, or it would have easier to find.

I was able to resolve the problem with `provider.AddTypeMapping<DateTimeOffset>("Date")`. What I really want is to remove the field that references the type derived from `IdentityUser`, so that the type isn't touched by introspection at all. However, that approach isn't working due to #120.
